### PR TITLE
Issue #43 fix: Added multiple entries while clicking on submit button multiple times & Edit own permission not working for draft record as per the requirement

### PR DESCRIPTION
**Pull request by vaibhavsTekdi**
**Steps to reproduce multiple entries submitted while clicking on submit/save as draft button multiple times.**

- Fill all the values for any UCM form & click multiple times on "Submit" / "Save as draft" button. It will save the record on each "Submit" / "Save as draft" button click.

**Steps to reproduce edit own permission issue on the draft record.**

1) Admin panel >> Create a UCM type with below permissions

   - Create Item: Allowed
   - View Item: Allowed
   - Edit Own Item: Allowed (Only owner of the record can edit)

2) User panel >> Save this form in draft mode.

3) Go to the UCM item list page of this UCM type & click on edit icon of that draft record.
    - Here, you will see a page with the message as "Item does not exist"

4) Go to the UCM item list page of this UCM type & click any record which will redirect to record detail page.
    - Here, you will see a page with the message as "Item does not exist"

### CODE REVIEW

File: site/views/itemform/tmpl/default.php [hunk 1]
```diff
@@ -193,12 +193,12 @@
 				if (!empty($this->allow_draft_save) && empty($itemState))
 				{
 					?>
-					<input type="button" class="btn btn-width150 br-0 btn-default font-normal" value="<?php echo JText::_("COM_TJUCM_SAVE_AS_DRAFT_ITEM"); ?>" onclick="steppedFormSave(this.form.id, 'draft');" />
+					<input type="button" class="btn btn-width150 br-0 btn-default font-normal" id="draftSave" value="<?php echo JText::_("COM_TJUCM_SAVE_AS_DRAFT_ITEM"); ?>" onclick="javascript: this.disabled=true; setTimeout(steppedFormSave, 1000, this.form.id, 'draft');" />
```
Comment (ankush-maherwal) [inline site/views/itemform/tmpl/default.php]: can't we disable the button until we get desired respnse from the ajax call?

Why we need to add a time out?

Comment (coolbung) [reply site/views/itemform/tmpl/default.php]: Absolutely. `setTimeout` is a strict no in this case. This needs to be improved to work based on the AJAX responses.

File: site/models/item.php [hunk 2]
```diff
@@ -62,6 +62,7 @@ protected function populateState()
 
 		// Check published state
 		if ((!$user->authorise('core.type.edititem', 'com_tjucm.type.' . $ucmId))
+			&& (!$user->authorise('core.type.editownitem', 'com_tjucm.type.' . $ucmId))
```
Comment (ankush-maherwal) [inline site/models/item.php]: do we need edit and editown both with &&

If user is allowed to edit the item but is not allowed to edit own item then the user will not be able to edit other users record

File: site/views/itemform/tmpl/default.php (hunk 1 above)
Comment (vaibhavsTekdi) [reply site/views/itemform/tmpl/default.php]: Now, these button will get enabled from ajax call response.

File: site/models/item.php (hunk 2 above)
Comment (vaibhavsTekdi) [reply site/models/item.php]: Have checked the above scenario & also different scenarios for this case & it works for both edit & edit own permission as per the  requirement.

**Pull request merged**